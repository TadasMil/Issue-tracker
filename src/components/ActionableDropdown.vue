<script setup lang="ts" name="ActionableDropdown">
const emit = defineEmits<{
  (event: 'edit'): void
  (event: 'mark-as-trashed'): void
  (event: 'mark-as-completed'): void
}>()
</script>

<template>
  <b-dropdown
    id="dropdown"
    aria-labelledby="dropdown"
    size="lg"
    variant="link"
    dropleft
    toggle-class="text-decoration-none"
    no-caret
  >
    <template #button-content>
      <div class="text-dark">
        <svg width="32" height="32" viewBox="0 0 32 32">
          <path
            fill="none"
            stroke="currentColor"
            stroke-width="2"
            d="M16 24a1 1 0 1 0 0-2a1 1 0 0 0 0 2Zm0-7a1 1 0 1 0 0-2a1 1 0 0 0 0 2Zm0-7a1 1 0 1 0 0-2a1 1 0 0 0 0 2Z"
          />
        </svg>
      </div>
    </template>

    <b-dropdown-item @click="emit('edit')">
      Edit
      <i class="bi bi-pencil fa-1x text-success"></i>
    </b-dropdown-item>

    <b-dropdown-item @click="emit('mark-as-trashed')">
      Mark as trashed
      <i class="bi bi-trash fa-1x text-danger"></i>
    </b-dropdown-item>

    <b-dropdown-item @click="emit('mark-as-completed')">
      Mark as completed
      <i class="bi bi-check-circle fa-1x text-success"></i>
    </b-dropdown-item>
  </b-dropdown>
</template>
