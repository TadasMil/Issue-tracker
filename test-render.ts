import type { RenderOptions } from '@testing-library/vue'
import BootstrapVue3 from 'bootstrap-vue-3'
import { render } from '@testing-library/vue'

const customRender = (ui: any, options?: RenderOptions) =>
  render(ui, {
    ...options,
    global: {
      ...options?.global,
      plugins: [BootstrapVue3, ...(options?.global?.plugins || [])],
    },
  })

export * from '@testing-library/vue'
export { customRender as render }
