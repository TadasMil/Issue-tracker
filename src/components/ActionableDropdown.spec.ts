import { describe, it, expect, vi } from 'vitest'
import { render, fireEvent } from '../../test-render'

import ActionableDropdown from './ActionableDropdown.vue'

vi.mock('vue-router', () => ({
  useRoute: vi.fn(() => ({ path: '/' })),
}))

describe('ActionableDropdown', () => {
  it('should render the component', () => {
    const { html } = render(ActionableDropdown)

    expect(html()).toMatchSnapshot()
  })

  it('should emit mark as trashed event', async () => {
    const { getByText, emitted } = render(ActionableDropdown)

    const button = getByText('Mark as trashed')

    await fireEvent.click(button)

    expect(emitted()).toHaveProperty('mark-as-trashed')
  })

  it('should emit mark as completed event', async () => {
    const { getByText, emitted } = render(ActionableDropdown)

    const button = getByText('Mark as completed')

    await fireEvent.click(button)

    expect(emitted()).toHaveProperty('mark-as-completed')
  })

  it('should emit edit event', async () => {
    const { getByText, emitted } = render(ActionableDropdown)

    const button = getByText('Edit')

    await fireEvent.click(button)

    expect(emitted()).toHaveProperty('edit')
  })
})
