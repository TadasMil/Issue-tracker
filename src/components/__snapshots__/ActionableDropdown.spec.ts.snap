// Vitest Snapshot v1

exports[`ActionableDropdown > should render the component 1`] = `
"<div class=\\"btn-group\\" aria-labelledby=\\"dropdown\\"><button class=\\"btn btn-link btn-lg text-decoration-none dropdown-toggle dropdown-toggle-no-caret\\" type=\\"button\\" id=\\"dropdown\\" data-bs-toggle=\\"dropdown\\" aria-expanded=\\"false\\">
    <!---->
    <div class=\\"btn-content\\">
      <div class=\\"text-dark\\"><svg width=\\"32\\" height=\\"32\\" viewBox=\\"0 0 32 32\\">
          <path fill=\\"none\\" stroke=\\"currentColor\\" stroke-width=\\"2\\" d=\\"M16 24a1 1 0 1 0 0-2a1 1 0 0 0 0 2Zm0-7a1 1 0 1 0 0-2a1 1 0 0 0 0 2Zm0-7a1 1 0 1 0 0-2a1 1 0 0 0 0 2Z\\"></path>
        </svg></div>
    </div>
  </button>
  <!---->
  <ul class=\\"dropdown-menu\\" aria-labelledby=\\"dropdown\\" role=\\"menu\\">
    <li role=\\"presentation\\" class=\\"\\"><button class=\\"dropdown-item\\" type=\\"button\\" target=\\"_self\\"> Edit <i class=\\"bi bi-pencil fa-1x text-success\\"></i></button></li>
    <li role=\\"presentation\\" class=\\"\\"><button class=\\"dropdown-item\\" type=\\"button\\" target=\\"_self\\"> Mark as trashed <i class=\\"bi bi-trash fa-1x text-danger\\"></i></button></li>
    <li role=\\"presentation\\" class=\\"\\"><button class=\\"dropdown-item\\" type=\\"button\\" target=\\"_self\\"> Mark as completed <i class=\\"bi bi-check-circle fa-1x text-success\\"></i></button></li>
  </ul>
</div>"
`;
